test/torch/tensors/test_private.py

```diff
@@ -0,0 +1,316 @@
+import pytest
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from syft.frameworks.torch.tensors.interpreters.private import PrivateTensor
+from syft.exceptions import GetNotPermittedError
+
+
+def test_wrap():
+    """
+    Test the .on() wrap functionality for LoggingTensor
+    """
+
+    x_tensor = torch.Tensor([1, 2, 3])
+    x = PrivateTensor().on(x_tensor)
+    assert isinstance(x, torch.Tensor)
+    assert isinstance(x.child, PrivateTensor)
+    assert isinstance(x.child.child, torch.Tensor)
+
+
+def test_native_private_tensor_method():
+    """
+    Test native's private_tensor method.
+    """
+    x_tensor = torch.Tensor([1, 2, 3])
+    private_x = x_tensor.private_tensor(allowed_users=("testing"))
+    assert isinstance(private_x, torch.Tensor)
+    assert isinstance(private_x.child, PrivateTensor)
+    assert isinstance(private_x.child.child, torch.Tensor)
+
+
+def test_allow_to_get():
+    """
+    Test native's private_tensor method.
+    """
+
+    x_tensor = torch.Tensor([1, 2, 3])
+
+    # User credentials mockup
+    class UserAuthMockup(object):
+        def __init__(self, login, password):
+            self.login = login
+            self.password = password
+
+        def __eq__(self, other):
+            if isinstance(other, UserAuthMockup):
+                return self.login == other.login and self.__password == other.password
+            else:
+                return False
+
+    registered_user = UserAuthMockup("user", "password")
+    registered_certificate = "encoded_certificate"
+    unregistered_user = UserAuthMockup("example", "password")
+
+    private_x = x_tensor.private_tensor(allowed_users=(registered_user, registered_certificate))
+    assert private_x.allowed_to_get(registered_user)
+    assert private_x.allowed_to_get(registered_certificate)
+    assert not private_x.allowed_to_get(unregistered_user)
+
+
+def test_get_method(workers):
+    bob = workers["bob"]
+    x_tensor = torch.Tensor([1, 2, 3])
+
+    private_x = x_tensor.private_tensor(allowed_users=("User"))
+
+    private_x_pointer = private_x.send(bob)
```

Comment:
Thinking about this, .send() should have all the same permissions as .get(), causing this to fail. Otherwise I could just try to call .send(me) to get around .get() blocking me.

Reply:
Yes, It makes sense! I'll fix it!

Reply:
Done!

Reply:
You're a rockstar.